File: test/conformance/host-fixtures/prewrapped_ok.lua
-- Host fixture implementation for the host-prewrapped-ok conformance
-- test. Both exports carry the exact emitted descriptor strings and
-- conforming behavior; ping's sync ->null return must be the __rt.__NULL
-- sentinel (enforced through the re-wrapped .f).

local rt = require("deal.runtime")

return {
  greet = {
    __kind = "function",
    sig = "(string)->string",
    f = function(name)
      return "hello " .. name
    end,
  },
  ping = {
    __kind = "function",
    sig = "()->null",
    f = function()
      return rt.__NULL
    end,
  },
}

File: test/conformance/host-fixtures/prewrapped_ok.js
"use strict";

// Host fixture implementation for the host-prewrapped-ok conformance
// test. The pre-wrapped sig-table form is a LuaJIT host-loader
// mechanism; the JS host realizes the declared-shape contract (the
// declared descriptor is the only trusted metadata): greet returns the
// declared string and ping's sync ->null return is the DEAL null, both
// enforced through the declared-descriptor wrapper.
module.exports = {
  greet: function (name) {
    return "hello " + name;
  },

  ping: function () {
    return null;
  },
};
